lib/publiccloud/basetest.pm

```diff
@@ -143,16 +144,20 @@ sub _cleanup {
     } else {
         diag('Public Cloud _cleanup: Not ready for provider cleanup.');
     }
-
-    _upload_logs();
 }
 
 sub _upload_logs {
     my ($self) = @_;
-    my @logs = ('/var/tmp/ssh_sut.log', '/var/log/cloudregister', '/etc/hosts', '/var/log/zypper.log', '/etc/zypp/credentials.d/SCCcredentials');
-    for my $logfile (@logs) {
-        script_run("sudo chmod a+r $logfile");
-        upload_logs($logfile, failok => 1, log_name => $logfile . ".txt");
+
+    my $ssh_sut_log = '/var/tmp/ssh_sut.log';
+    script_run("sudo chmod a+r " . $ssh_sut_log);
+    upload_logs($ssh_sut_log, failok => 1, log_name => $ssh_sut_log . ".txt");
+    return unless $self->{run_args} && $self->{run_args}->{my_instance};
+
+    my @instance_logs = ('/var/log/cloudregister', '/etc/hosts', '/var/log/zypper.log', '/etc/zypp/credentials.d/SCCcredentials');
+    for my $instance_log (@instance_logs) {
+        $self->{run_args}->{my_instance}->ssh_script_run("sudo chmod a+r " . $instance_log);
+        $self->{run_args}->{my_instance}->upload_log($instance_log, failok => 1, log_name => $instance_log . ".txt");
```

Comment:
PLease pay attention here, the `upload_log` is different from the `upload_logs` in original vesrion. In fact the 1st needs the instance pointer and includes a `scp` file transfter from cloud to local helper image. The 2nd instead is the original [openqa upload log](https://open.qa/api/testapi/#_upload_logs), without scp and not neeeding instance pointer.
Telling this 'cause when the cleanup is called _after_ `ssh_interactive_start` and before terraform destroy, you still are on the VM and upload context is different.

Reply:
Thanks for reviewing. That's exactly the problem which this PR addresses! The goal is to upload the SUT logs (PC instance).